### PR TITLE
Markdown docs

### PR DESCRIPTION
**Pull request by esoergel**
Add markdown support to our readthedocs docs.  I do think we should use rst first and foremost, but I'd rather have docs in one place, and this seems to be a barrier, so this can be a transition plan.

@czue @calellowitz @orangejenny 

image

### CODE REVIEW

File: requirements/dev-requirements.in [hunk 1]
```diff
@@ -16,3 +16,4 @@ modernize
 Fabric==1.14.0
 gnureadline==6.3.8
 pip-tools~=3.0
+recommonmark==0.5.0
```
Comment (esoergel) [inline requirements/dev-requirements.in]: Should `dev-requirements` import all from `docs-requirements`?

**Review by emord — APPROVED** [review]

File: requirements/dev-requirements.in (hunk 1 above)
Comment (esoergel) [reply requirements/dev-requirements.in]: Good point - I can't think of a good reason to ever pin this particular requirement.  I'll drop that, move it to docs-requirements, and import that here.

Comment (esoergel) [reply requirements/dev-requirements.in]: punting on importing from `docs-requirements` - it also imports from requirements, and I get pip-compile errors about installing from github without `-e`

Comment (emord) [reply requirements/dev-requirements.in]: yeah. i thought that may be the case. seems fine to punt for now

**Comment by millerdev** [comment]
> I do think we should use rst first and foremost, but I'd rather have docs in one place, and this seems to be a barrier, so this can be a transition plan.

So does this mean

- _If you know how to write reStructureText, do that. But if you don't or don't want to please use markdown rather than skipping documentation altogether._

Or maybe

- _Feel free to use reStructureText or markdown when writing documentation. We may eventually convert it all to one or the other._

Or something else?

**Comment by emord** [comment]
My vote is to allow some free form documentation until we get more docs. If we find that we want to standardize, we can go back and convert them (I used `pandoc --from=markdown --to=rst --output=README.rst README.md` last night and it worked well for userreports readme which is probably our largest one)

**Comment by esoergel** [comment]
I'm fine with a mix of both, honestly - it's not that much overhead.  I imagine many things will start off as markdown, then maybe get converted to rst when someone wants to do something complex with them.  As Emord said, switching is pretty straightforward.

**Comment by czue** [comment]
this is great! thanks for doing it.

**Pull request merged**